h/services/user_update.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from sqlalchemy.exc import SQLAlchemyError
+
+from h.services.exceptions import ConflictError, ValidationError
+
+
+class UserUpdateService(object):
+    def __init__(self, session):
+        """
+        Create a new UserUpdateService
+
+        :param session: the SQLAlchemy session object
+        """
+        self.session = session
+
+    def update(self, user, **kwargs):
+        """
+        Update a user model with the args provided.
+
+        :arg user: the group to update
+        :type user: ~h.models.User
+
+        :raise ValidationError: if setting an attribute on the model raises :exc:`ValueError`
+                                or if ``authority`` is present in ``kwargs``
+
+        :rtype: ~h.models.User
+        """
+
+        # Much repurcussion if a user's authority is changed at this point.
+        # May wish to re-evaluate later if users need to be moved between
+        # authorities.
+        if "authority" in kwargs:
+            raise ValidationError("A user's authority may not be changed")
+
+        for key, value in kwargs.items():
+            try:
+                setattr(user, key, value)
+            except ValueError as err:
+                raise ValidationError(err)
+
+        try:
+            self.session.flush()
+
+        except SQLAlchemyError as err:
+            # Handle DB integrity issues with duplicate ``authority_provided_id``
+            if (
+                'duplicate key value violates unique constraint "ix__user__userid"'
```

Comment:
This is potentially brittle, in case the message ever changes in future. I see we have a test that would catch such a scenario though provided SQLAlchemy in dev is kept sufficiently in sync with what we use in QA/prod.

Reply:
Yes, it feels just a tad so to me, as well. I was following the established pattern in the group update service, but I'll talk to Sean about it.